sympy/physics/continuum_mechanics/cross_section.py

```diff
@@ -0,0 +1,161 @@
+from sympy.core import sympify
+from sympy.printing import sstr
+from sympy.core import Symbol, S
+
+
+class CrossSection(object):
+    def __init__(self, shape, variable, *dimensions):
+        self.shape = shape
+        self.dimensions = dimensions
+        self.variable = variable
+
+        shape = shape.lower()
+        
+        if shape == "circular":
+            self._circular(self.dimensions)
+        elif shape == "triangular":
+            self._triangular(self.dimensions)
+        elif shape == "rectangular":
+            self._rectangular(self.dimensions)
+        elif shape == "tsection":
+            self._tsection(self.dimensions)
+        elif shape == "isection":
+            self._isection(self.dimensions)
+        else:
+            raise ValueError("No such cross-sectional shape is supported")
+
+    def __str__(self):
+        str_1 = 'CrossSection({}, {})'.format(sstr(self._shape), sstr(self._dimensions))
+        return str_1
+
+    __repr__ = __str__
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @shape.setter
+    def shape(self, shape):
+        self._shape = shape
+    
+    @property
+    def dimensions(self):
+        return self._dimensions
+
+    @dimensions.setter
+    def dimensions(self, dim):
+        self._dimensions = dim
+    
+    @property
+    def variable(self):
+        return self._variable
+    
+    @variable.setter
+    def variable(self, var):
+        if isinstance(var, Symbol):
+            self._variable = var
+        else:
+            raise TypeError("""The variable should be a Symbol object.""")
+
+    @property
+    def area(self):
+       return self._area
+
+    @property
+    def centroid(self):
+       return self._centroid
+    
+    @property
+    def second_moment(self):
+       return self._second_moment
+            
+    @property
+    def section_modulus(self):
+       return self._section_modulus
+               
+    @property
+    def polar_modulus(self):
+       return self._polar_modulus
+
+
+    def _circular(self, radius):
+        radius = radius[0]
+        circle = Circle(Point(radius, radius), radius)
+
+        self._area = circle.area
+        self._centroid = circle.center
+        self._second_moment = circle.second_moment_of_area()
+        self._section_modulus = self._second_moment[0]/(self.variable - self.centroid.y)
+        self._polar_modulus = self._second_moment[0] + self._second_moment[1]
+
+
+    def _triangular(self, dimensions):
+        width, height = dimensions
+        triangle = Polygon((0,0), (width, 0), (width/2, height))
+
+        self._centroid = traingle.centroid
+        self._area = traingle.area
+        self._second_moment = traingle.second_moment_of_area()
+        self._centroid = traingle.centroid
+        self._section_modulus = self._second_moment[0]/(self.variable - self._centroid.y)
+        self._polar_modulus = self._second_moment[0] + self._second_moment[1]
```

Comment:
This duplicates the lines above. It would be better to let the user provide a triangle or circle shape as input. Then you can just call all these methods on self.shape.